File: backend/tour-design-api/src/main/java/com/tourdesign/platform/service/PackageHotelService.java
package com.tourdesign.platform.service;

import com.tourdesign.platform.model.PackageHotelModel;
import java.util.List;
import java.util.Optional;

public interface PackageHotelService {

    List<PackageHotelModel> list();

    PackageHotelModel create(PackageHotelModel obj);

    Optional<PackageHotelModel> search(Long id);

    List<PackageHotelModel> findByTravelPackageId(Long packageId);

    Optional<PackageHotelModel> update(Long id, PackageHotelModel obj);

    boolean delete(Long id);
}

File: frontend/tour-design-web/src/app/services/package-hotel.service.ts
import { Injectable } from '@angular/core';
import { ApiService } from './api.service';
import { PackageHotelModel } from '../models/package-hotel.model';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class PackageHotelService {
  private baseUrl = '/package-hotel';

  constructor(private api: ApiService) {}

  list(): Observable<PackageHotelModel[]> {
    return this.api.get<PackageHotelModel[]>(this.baseUrl);
  }

  search(id: number): Observable<PackageHotelModel> {
    return this.api.get<PackageHotelModel>(`${this.baseUrl}/${id}`);
  }

  create(obj: PackageHotelModel): Observable<PackageHotelModel> {
    return this.api.post<PackageHotelModel>(this.baseUrl, obj);
  }

  update(id: number, obj: PackageHotelModel): Observable<PackageHotelModel> {
    return this.api.put<PackageHotelModel>(`${this.baseUrl}/${id}`, obj);
  }

  delete(id: number): Observable<void> {
    return this.api.delete<void>(`${this.baseUrl}/${id}`);
  }
}